APISamples/typescript/src/api.ts

```diff
@@ -0,0 +1,170 @@
+import { CatalogItem } from './api';
+import axiosLib, { AxiosInstance } from "axios";
+export class Api {
+    private serverUrl:string;
+    private axios:AxiosInstance;
+
+    constructor(server:string = 'http://localhost/reports'){
+        this.axios = axiosLib.create({
+            baseURL: `${server}/api/v2.0`,
+            withCredentials: true,
+            headers: {
+                'Content-Type':'application/json;charset=UTF-8'
+            }
+        });
+    }
+
+    requestOptions(method:string, body?:any): RequestInit {
+        return {
+            credentials: 'include',
+            mode: 'cors',
+            headers: {
+                'Content-Type':'application/json;charset=UTF-8'
+            },
+            method,
+            body: JSON.stringify(body)
+        }
+    }
+
+    async getFolderItemsAsync(path:string = '/') : Promise<CatalogItem[] | CatalogItem> {
+        let response = await this.axios.get(`/Folders(Path='${path}')/CatalogItems`);
+        let items = response.data;
+        return items.value;
+    }
+
+    async postCatalogItemAsync(item:CatalogItem) : Promise<void> {
+        let url: string = `${this.serverUrl}/CatalogItems`;
+        let response = await this.axios.post('/CatalogItems',item);
+        let createdItem = response.data;
+        return createdItem;
+    }
+
+    async meAsync() : Promise<User> {
+        let url: string = `${this.serverUrl}/me`;
+        let response = await this.axios.get('/me');
+        let item = response.data
+        return item;
+    }
+
+    async uploadFileAsync(file: File) : Promise<void> {
+        let fileInfo = await this.getFileInfo(file);
+        let path = window.location.hash.substring(1);
+        path = path.length > 1 ? path : '';
+        let item:any = {
+            Content: fileInfo.content,
+            Path: `${path}/${fileInfo.name}`,
+            Name: fileInfo.name,
+            ContentType: fileInfo.contentType
+        }
+
+        switch (fileInfo.extension.toLocaleLowerCase()) {
+            case 'rdl':
+                item['@odata.type'] = '#Model.Report';
+                break;
+            case 'rsd':
+                item['@odata.type'] = '#Model.DataSet';
+                break;
+            case 'rds':
+                item.ContentType = 'text/xml';
+                item['@odata.type'] = '#Model.Resource';
```

Comment:
Shouldn't this be DataSource?

Reply:
DataSources can't be uploaded, they need to be created using the API